protocols/luci-proto-yggdrasil/htdocs/luci-static/resources/protocol/yggdrasil.js

```diff
@@ -0,0 +1,272 @@
+'use strict';
+'require form';
+'require network';
+'require rpc';
+'require tools.widgets as widgets';
+'require uci';
+'require ui';
+network.registerPatternVirtual(/^yggdrasil-.+$/);
+
+function validatePrivateKey(section_id,value) {
+	if (value.length == 0) {
+		return true;
+	};
+	if (!value.match(/^([0-9a-fA-F]){128}$/)) {
+		if (value != "auto") {
+			return _('Invalid private key string');
+		}
+		return true;
+	}
+	return true;
+};
+
+function validatePublicKey(section_id,value) {
+	if (value.length == 0) {
+		return true;
+	};
+	if (!value.match(/^([0-9a-fA-F]){64}$/))
+		return _('Invalid public key string');
+	return true;
+};
+
+function validateYggdrasilListenUri(section_id,value) {
+	if (value.length == 0) {
+		return true;
+	};
+	if (!value.match(/^(tls|tcp|unix|quic):\/\//))
+		return _('URI scheme not supported');
+	return true;
+};
+
+function validateYggdrasilPeerUri(section_id,value) {
+	if (!value.match(/^(tls|tcp|unix|quic|socks|socktls):\/\//))
+		return _('URI scheme not supported');
+	return true;
+};
+
+var cbiKeyPairGenerate = form.DummyValue.extend({
+	cfgvalue: function(section_id, value) {
+		return E('button', {
+			'class':'btn',
+			'click':ui.createHandlerFn(this, function(section_id,ev) {
+				var prv = this.section.getUIElement(section_id,'private_key'),
+					pub = this.section.getUIElement(section_id,'public_key'),
+					map = this.map;
+				if ((prv.getValue()||pub.getValue()) && !confirm(_('Do you want to replace the current keys?')))
```

Comment:
Remove this confirmation: the user has the opportunity to review what changes will be applied as a whole when they go to apply. Make changing the config friction free.

Reply:
bit the bullet, removed public_key option field since it cannot be obtained during generation. its also ignored by the daemon, but.. it is sort-of important and we will have to raise a feature request to yggdrasil devs. until then, its removed.

removed the confirm popup, works fine.